xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-profile/xwiki-platform-user-profile-test/xwiki-platform-user-profile-test-pageobjects/src/main/java/org/xwiki/user/test/po/PreferencesUserProfilePage.java

```diff
@@ -33,13 +33,13 @@ public class PreferencesUserProfilePage extends AbstractUserProfilePage
     @FindBy(xpath = "//div[@id='preferencesPane']//div[@class='editProfileCategory']/a")
     private WebElement editPreferences;
 
-    @FindBy(xpath = "//div[@id='preferencesPane']/div[1]/div/dl[2]/dd[1]")
+    @FindBy(xpath = "//div[@id='preferencesPane']/div[1]/div/dl[3]/dd[1]")
```

Comment:
Why keeping fragile and hard to read selectors like this? 
Afaics all the `dd` elements have a nice `data-user-property="TYPE"` that we can use to define much better selectors.

Reply:
I'm not sure how to test this exactly yet (without spending half an hour applying the changes from XWIKI-21492 manually). So I stuck with the simplest solution that I knew would not create a regression. I'm not super comfortable with xpaths yet so I'd want to test if I made a larger change.
If we don't merge this right now, I'll wait for the updated build tomorrow, test with it and update the PR then.

Reply:
You can quite easily build the changes locally and run the tests again.
Something like `mvn clean install -Pintegration-tests,docker -pl :xwiki-platform-user-profile-ui,:xwiki-platform-user-profile-test-pageobjects,:xwiki-platform-user-profile-test-docker` should allow you to test you change (assuming your branch is up-to-date with master).

Reply:
You can even add the `-U` option to also update unrelated dependencies, but with the set of modules listed in `-pl`, all the relevant bits should be ok `mvn clean install -Pintegration-tests,docker -pl :xwiki-platform-user-profile-ui,:xwiki-platform-user-profile-test-pageobjects,:xwiki-platform-user-profile-test-docker`
Also, if you want to speed up the docker validation part, you can filter on a given test:
`mvn clean install -Pintegration-tests,docker -pl :xwiki-platform-user-profile-ui,:xwiki-platform-user-profile-test-pageobjects,:xwiki-platform-user-profile-test-docker -Dit.test=MyClass#mymethod`.
Finally, there is also an option to run the test against a server running on localhost:8080 (see https://dev.xwiki.org/xwiki/bin/view/Community/Testing/DockerTesting/#HExistingXWikiinstance)